api/types/user.go

```diff
@@ -450,6 +452,16 @@ func (u *UserV2) ResetLocks() {
 	u.Spec.Status.RecoveryAttemptLockExpires = time.Time{}
 }
 
+// DeepCopy creates a clone of this user value.
+func (u *UserV2) DeepCopy() User {
+	// github.com/golang/protobuf/proto.Clone panics when trying to
+	// copy a map[K]V where the type of V is a slice of anything
+	// other than byte. See https://github.com/gogo/protobuf/issues/14
+	uV2 := protoadapt.MessageV2Of(u)
+	uV2Copy := proto.Clone(uV2)
+	return protoadapt.MessageV1Of(uV2Copy).(User)
```

Comment:
I believe we're guaranteed to get back a `*UserV2` in this, and I'd rather have this than get back something that accidentally implements `User`.

```suggestion
	return protoadapt.MessageV1Of(uV2Copy).(*UserV2)
```

Reply:
Done